compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -944,6 +947,69 @@ LogicalResult setScatterLoweringConfig(IREE::GPU::TargetAttr target,
       {flatWorkgroupSize, 1, 1}, flatWorkgroupSize, DictionaryAttr());
 }
 
+//====---------------------------------------------------------------------===//
+// Sort Pipeline Configuration
+//====---------------------------------------------------------------------===//
+
+LogicalResult setSortConfig(IREE::GPU::TargetAttr target,
+                            mlir::FunctionOpInterface entryPoint,
+                            Operation *op) {
+  auto context = op->getContext();
+  Builder b(context);
+
+  auto subgroupSize = target.getPreferredSubgroupSize();
+  TileSizesListType tileSizes;
+  auto interfaceOp = cast<PartitionableLoopsInterface>(*op);
+  auto partitionedLoops =
+      interfaceOp.getPartitionableLoops(kNumMaxParallelDims);
+
+  auto createLoweringConfig = [&](SmallVector<int64_t> workgroupSizes,
+                                  SmallVector<int64_t> threadSizes) {
+    SmallVector<NamedAttribute, 2> attrs = {
+        NamedAttribute("workgroup", b.getI64ArrayAttr(workgroupSizes)),
+        NamedAttribute("thread", b.getI64ArrayAttr(threadSizes))};
+    auto configDict = b.getDictionaryAttr(attrs);
+    return IREE::GPU::LoweringConfigAttr::get(context, configDict);
+  };
+
+  if (partitionedLoops.empty()) {
+    auto loweringConfig = createLoweringConfig({1, 1, 1}, {1, 1, 1});
+    return setOpConfigAndEntryPointFnTranslation(
+        entryPoint, op, loweringConfig, CodeGenPipeline::LLVMGPUTileAndFuse,
+        {1, 1, 1}, subgroupSize, DictionaryAttr());
+  }
+
+  size_t numLoops = partitionedLoops.back() + 1;
+
+  // To get peak occupancy we need a workgroup size of at least two warps
+  std::array<int64_t, 3> workgroupSize = {2 * subgroupSize, 1, 1};
+  SmallVector<int64_t> workgroupTileSizes(numLoops, 1);
+  SmallVector<int64_t> threadTileSizes(numLoops, 1);
+
+  // Set all non-parallel loops to zero tile size.
+  llvm::DenseSet<unsigned> partitionedLoopsSet(partitionedLoops.begin(),
+                                               partitionedLoops.end());
+  for (auto depth : llvm::seq<int64_t>(0, numLoops)) {
+    if (!partitionedLoopsSet.count(depth)) {
+      workgroupTileSizes[depth] = 0;
+    }
+  }
+
+  // Tile to have one element per thread.
+  for (int64_t depth = numLoops; depth > 0; depth--) {
```

Comment:
```suggestion
  for (int64_t depth = numLoops; depth > 0; --depth) {
```
https://llvm.org/docs/CodingStandards.html#prefer-preincrement